src/github.com/matrix-org/go-neb/services/rssbot/rssbot_test.go

```diff
@@ -103,3 +116,55 @@ func TestHTMLEntities(t *testing.T) {
 	// Check that the Matrix client sent a message
 	wg.Wait()
 }
+
+func TestFeedItemFiltering(t *testing.T) {
+	feedURL := "https://thehappymaskshop.hyrule"
+
+	// Create rssbot client
+	rssbot, err := createRSSClient(feedURL)
+	if err != nil {
+		t.Fatal("Failed to create RSS bot: ", err)
+	}
+
+	feed := rssbot.Feeds[feedURL]
+	feed.MustInclude.Title = []string{"Zelda"}
+	rssbot.Feeds[feedURL] = feed
+
+	_, items, _ := rssbot.queryFeed(feedURL)
+	// Expect that we get no items if we filter for 'Zelda' in title
+	if len(items) != 0 {
+		t.Errorf("Expected 0 items, got %v", items)
+	}
+
+	// Recreate rssbot client
+	rssbot, err = createRSSClient(feedURL)
+	if err != nil {
+		t.Fatal("Failed to create RSS bot: ", err)
+	}
+
+	feed = rssbot.Feeds[feedURL]
+	feed.MustInclude.Title = []string{"Majora"}
+	rssbot.Feeds[feedURL] = feed
+
+	_, items, _ = rssbot.queryFeed(feedURL)
+	// Expect one item if we filter for 'Majora' in title
+	if len(items) != 1 {
+		t.Errorf("Expected 1 item, got %d", len(items))
+	}
+
+	// Recreate rssbot client
+	rssbot, err = createRSSClient(feedURL)
+	if err != nil {
+		t.Fatal("Failed to create RSS bot: ", err)
+	}
+
+	feed = rssbot.Feeds[feedURL]
+	feed.MustNotInclude.Author = []string{"kid"}
```

Comment:
That's an interesting edge case. I'm not sure we want to do that. We probably want to treat each item as a word rather than a partial string so the filter "Jon" catches "Jon Snow" but not "Jonathan Creek".

Reply:
I see. Changed the code to split the string into words (delimited by non-alphanumeric) and the documentation to reflect that.